### PR TITLE
feat: auth functionality added using redux

### PR DESCRIPTION
**Pull request by pankaj-prog**

### CODE REVIEW

**Review by SinchanaMS — APPROVED** [review]
Very neatly written code. I couldn't find any issues.
Great work!!

File: src/App.js [hunk 1]
```diff
@@ -27,20 +30,23 @@ function App() {
...
-      </Route>
-      <Route path="*" element={<Page404 />} />
-    </Routes>
+    <>
+      <Routes>
+        <Route path="/" element={<UserRoutes />}>
+          <Route path="home" element={<Home />} />
+          <Route path="profile" element={<Profile />} />
+          <Route path="explore" element={<Explore />} />
+          <Route path="post/:postID" element={<PostPage />} />
```
Comment (tarunsankhla) [inline src/App.js:39]: use constant instead of direct string

File: package.json [hunk 2]
```diff
@@ -16,8 +17,10 @@
     "react-dom": "^17.0.2",
     "react-drag-drop-files": "^2.3.5",
     "react-icons": "^4.3.1",
+    "react-redux": "^8.0.1",
     "react-router-dom": "^6.3.0",
     "react-scripts": "4.0.3",
+    "react-toastify": "^9.0.1",
     "uuid": "^8.3.2",
```
Comment (manojsarna) [inline package.json:24]: using react toastify, exploring new packages is good

File: src/App.js [hunk 3]
```diff
@@ -27,20 +30,23 @@ function App() {
...
+          <Route path="newpost" element={<NewPost />} />
+        </Route>
```
Comment (manojsarna) [inline src/App.js:41]: Whats the requirement of new route for new post?

File: src/pages/auth/Signup.jsx [hunk 4]
```diff
@@ -38,16 +79,26 @@ const SignUp = () => {
...
             <label htmlFor="checkbox">
               I’m in for emails with exciting discounts and personalized
               recommendations
             </label>
           </div>
-          <button type="button" className="auth-btn form-btn btn-rc">
+          <button
+            type="button"
+            className="auth-btn form-btn btn-rc"
+            onClick={(e) => signUpHandler(e)}
```
Comment (tarunsankhla) [inline src/pages/auth/Signup.jsx]: no need to pass event

File: src/pages/auth/Login.jsx [hunk 5]
```diff
@@ -1,52 +1,111 @@
...
+  const { isLoggedIn } = useSelector((state) => state.auth);
+
+  const loginClickHandler = () => {
+    setIsPasswordValid(true);
+    if (validatePassword(passwordRef.current.value)) {
+      dispatch(
+        loginUser({
+          username: usernameRef.current.value,
+          password: passwordRef.current.value,
+        })
```
Comment (manojsarna) [inline src/pages/auth/Login.jsx:25]: Appreciation : good use of useRef

File: src/redux/reducers/authSlice.js [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+    }
+  }
+);
+
+// handles signup
+export const signUpUser = createAsyncThunk(
+  "auth/signUpUser",
+  async (signUpData, { rejectWithValue }) => {
+    try {
+      const { data } = axios.post("/api/auth/signup", signUpData);
```
Comment (tarunsankhla) [inline src/redux/reducers/authSlice.js:31]: create constant for api string

File: src/redux/reducers/authSlice.js [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+      localStorage.setItem("buzzgram-user", JSON.stringify(data.foundUser));
+      localStorage.setItem("buzzgram-token", data.encodedToken);
```
Comment (tarunsankhla) [inline src/redux/reducers/authSlice.js:33]: same for this make variable and use string

**Review by tarunsankhla — APPROVED** [review]
over code is great appreciation for thunk

File: src/redux/reducers/authSlice.js [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+      return data;
+    } catch (err) {
+      return rejectWithValue(err?.response);
+    }
+  }
+);
+
```
Comment (manojsarna) [inline src/redux/reducers/authSlice.js:40]: Good work with the AuthSlice

**Review by manojsarna — APPROVED** [review]
Overall the code is good!!
 Check the suggestions

File: src/App.js (hunk 3 above)
Comment (pankaj-prog) [reply src/App.js:41]: for creating a new post

**Pull request merged**